gammapy/datasets/io.py

```diff
@@ -294,12 +294,12 @@ def get_valid_path(self, filename):
         Parameters
         ----------
         filename : str or `Path`
```

Comment:
If we want `Path` to be properly referred to and not just highlighted in blue, we should pu the full name:

```suggestion
        filename : str or `~pathlib.Path`
```

There are multiple case of this.